Clean/Gadgets/BLAKE3/ApplyRounds.lean

```diff
@@ -734,9 +734,18 @@ theorem completeness : Completeness (F p) elaborated Assumptions := by
   · apply h_state_normalized
   · apply h_message_normalized
 
-
+-- Unfortunately @[simps! (config := {isSimp := false, attrs := [`circuit_norm]})] timeouts.
+-- Therefore I had to add simplification rules `circuit_assumptions_is` and `circuit_spec_is` manually.
 def circuit : FormalCircuit (F p) Inputs BLAKE3State := {
   elaborated with Assumptions, Spec, soundness, completeness
 }
 
+@[circuit_norm]
+lemma circuit_assumptions_is :
+  circuit.Assumptions (F := F p) = Assumptions := rfl
+
+@[circuit_norm]
+lemma circuit_spec_is :
+  circuit.Spec (F := F p) = Spec := rfl
```

Comment:
these should definitely not go into `circuit_norm`, they are used in a single other proof and can just be added there directly

in general, anything about specific gadgets should be considered in user land and should not modify the core library. the only exception are core, very frequently used gadgets like `assertBool` or `Equality.circuit`